### PR TITLE
BAU - refactored Title to ensure that translated messages are not used

### PR DESCRIPTION
**Pull request by SteveSugden**
Title translates all parts of the title and as all view tests extend
ViewSpec (which uses 'AllMessageKeysAreMandatoryMessages') then the
message keys are validated to exist in the messages file.

### CODE REVIEW

File: test/unit/views/movement/ConsignmentReferenceViewSpec.scala [hunk 1]
```diff
@@ -31,7 +31,7 @@ class ConsignmentReferenceViewSpec extends ViewSpec {
 
     "render title" in {
 
-      page(ConsignmentReferences.form()).getTitle must containMessage("consignmentReferences.title")
+      page(ConsignmentReferences.form()).getTitle must containMessage("consignmentReferences.ARRIVE.question")
```
Comment (PatrykRudnicki) [inline test/unit/views/movement/ConsignmentReferenceViewSpec.scala]: I know it's not a part of this PR but I would prefer to have `arrive` not `ARRIVE`

Comment (SteveSugden) [reply test/unit/views/movement/ConsignmentReferenceViewSpec.scala]: Normally I would point out that we have used enum raw values in a few places as part of the message keys.  But as we have time atm I'll change it.   
Just don't expect this level of service all of the time.

File: app/views/Title.scala [hunk 2]
```diff
@@ -18,13 +18,16 @@ package views
 
 import play.api.i18n.Messages
 
-case class Title(title: String)
+case class Title(headingKey: String, sectionKey: Option[String] = None, headingArgs: Seq[String] = Seq.empty) {
```
Comment (bambuchaAdm) [inline app/views/Title.scala]: What do you think about use empty string as null object? code would be simple - less pointless `Some` in code

Comment (SteveSugden) [reply app/views/Title.scala]: I guess so.  Defaulted to "".

**Pull request merged**